pulsar-broker/src/test/java/org/apache/pulsar/broker/service/persistent/PersistentTopicTest.java

```diff
@@ -577,15 +586,21 @@ public void testCreateTopicWithZombieReplicatorCursor(boolean topicLevelPolicy)
         assertEquals(getCursors.get(), Collections.singleton(conf.getReplicatorPrefix() + "." + remoteCluster));
 
         if (topicLevelPolicy) {
-            admin.topics().setReplicationClusters(topicName, Collections.emptyList());
+            admin.topics().setReplicationClusters(topicName, Collections.singletonList("test"));
         } else {
-            admin.namespaces().setNamespaceReplicationClustersAsync(namespace, Collections.emptySet()).get();
+            admin.namespaces().setNamespaceReplicationClustersAsync(namespace, Collections.singleton("test")).get();
         }
         admin.clusters().deleteCluster(remoteCluster);
         // Now the cluster and its related policy has been removed but the replicator cursor still exists
 
-        topic.initialize().get(3, TimeUnit.SECONDS);
-        Awaitility.await().atMost(3, TimeUnit.SECONDS)
-                .until(() -> !topic.getManagedLedger().getCursors().iterator().hasNext());
+        // Verify:
+        // 1. Topic can load success( use "topic.initialize()" instead of "load topic" ).
+        //    If the topic loading by client is failed, it will retry, so we can do retry "topic.initialize()".
+        // 2. The repl cursor will be deleted.
+        Awaitility.await().atMost(10, TimeUnit.SECONDS)
+                .until(() -> {
+                    topic.initialize().get(3, TimeUnit.SECONDS);
```

Comment:
It's meaningless because `topic.initialize` might throw an exception and it won't be retried.

```

java.util.concurrent.ExecutionException: java.lang.RuntimeException: org.apache.pulsar.metadata.api.MetadataStoreException$NotFoundException: remote

	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2096)
	at org.apache.pulsar.broker.service.persistent.PersistentTopicTest.lambda$testCreateTopicWithZombieReplicatorCursor$10(PersistentTopicTest.java:602)
	at org.awaitility.core.CallableCondition$ConditionEvaluationWrapper.eval(CallableCondition.java:99)
```

Reply:
perhaps `.ignoreExceptionsInstanceOf(MetadataStoreException.NotFoundException.class)` could help?

Reply:
BTW, the root cause is not here. I added a `try-catch` block to swallow the exception, the test could still fail.

```java
            try {
                topic.initialize().get(3, TimeUnit.SECONDS);
            } catch (ExecutionException e) {
                log.warn("Failed to initialize: {}", e.getCause().getMessage());
            }
```

Reply:
> BTW, the root cause is not here. I added a try-catch block to swallow the exception, the test could still fail.

The root cause is like this:
1. a replication task starting
2. failed to get cluster meta from ZK
3. check replication policies
  - Since the policy is updated asynchronously, the cache has not been updated.
  - get clusters( responses `[local_cluster, remote_cluster]`)
4. Since check replication successes, will not delete the cursor. bingo

Reply:
I've run this test 4*200 times with no errors

![截屏2023-04-06 21 04 25](https://user-images.githubusercontent.com/25195800/230387092-f2dab624-99e3-4ca7-ab81-39678fb2445d.png)

Reply:
>  just mock the load operation by client,

Actually, the operation failed at `topic.initialize` in my local env. An exception will be thrown and it won't retry.

Reply:
See step 4 above, it goes to the logic branch `else`, and the error is thrown.

https://github.com/apache/pulsar/pull/20025/files#diff-5edf14cc6f25857d0cfdd26b2d3b3141230ecfb0dfa95aebf7583fd76ede4c4bR1748-R1751

Reply:
I have change the test to `Awaitility.await().atMost(10, TimeUnit.SECONDS).ignoreExceptions().untilAsserted(`, it will be fixed